packages/notification-center/src/components/notification-center/components/user-preference/SubscriberPreference.tsx

```diff
@@ -17,13 +17,19 @@ export function SubscriberPreference() {
   const { setFetchingStrategy } = useNovuContext();
   const { theme, common } = useNovuTheme();
   const { data, isLoading: arePreferencesLoading } = useFetchUserPreferences();
+  const { preferenceFilter } = useNotificationCenter();
   const [rootStyles, accordionItemStyles, accordionContentStyles, accordionControlStyles, accordionChevronStyles] =
     useStyles(['preferences.root', 'accordion.item', 'accordion.content', 'accordion.control', 'accordion.chevron']);
 
   const baseTheme = theme?.userPreferences;
-  const preferences = data
+  let preferences = data
     ?.filter((item) => !item.template.critical)
     ?.filter((pref) => Object.keys(pref.preference.channels).length > 0);
+
+  if (preferenceFilter && preferences) {
```

Comment:
here we use the filter function.
added it under the if statement and not under nested filtering above because in most cases we the user won't provide a filter function and the iteration will be refunded.